caas/kubernetes/provider/specs/v3.go

```diff
@@ -296,11 +296,26 @@ func (w K8sValidatingWebhookSpec) Validate() error {
 	return nil
 }
 
+// K8sService is a subset of v1.Service which defines
+// attributes we expose for charms to set.
+type K8sService struct {
```

Comment:
Can we just call this "Service" instead of "K8sService" ?
eg we don't have "K8sSecret" but "Secret"